launcher/minecraft/mod/tasks/LocalResourcePackParseTask.cpp

```diff
@@ -178,6 +179,168 @@ bool processZIP(ResourcePack& pack, ProcessingLevel level)
+            result = true;
+            return true;
+        } else if (bool_str == "false") {
+            result = false;
+            return true;
+        } else {
+            qWarning() << "Invalid bool value in component!";
+            return false;
+        }
+    } else {
+        qWarning() << "Invalid type where bool expected!";
+        return false;
+    }
+}
+
+bool readFormat(const QJsonObject& obj, TextFormat& format)
+{
+    auto text = obj.value("text");
+    auto color = obj.value("color");
+    auto bold = obj.value("bold");
+    auto italic = obj.value("italic");
+    auto underlined = obj.value("underlined");
+    auto strikethrough = obj.value("strikethrough");
+    auto extra = obj.value("extra");
+
+    if (color.isString()) {
+        // colors can either be a hex code or one of a few text colors
+        auto col_str = color.toString();
+
+        const QRegularExpression hex_expression("^#([a-fA-F0-9]{6}|[a-fA-F0-9]{3})$");
+        const auto hex_match = hex_expression.match(col_str);
+
+        if (hex_match.hasMatch()) {
+            format.color = color.toString();
+        } else {
+            if (!textColorToHexColor(color.toString(), format.color)) {
+                qWarning() << "Invalid color type in component!";
+                return false;
+            }
+        }
+    }
+
+    return getBoolOrFromText(bold, format.bold) && getBoolOrFromText(italic, format.italic) &&
+           getBoolOrFromText(underlined, format.underlined) && getBoolOrFromText(strikethrough, format.strikethrough);
+}
+
+void appendBeginFormat(TextFormat& format, QString& toAppend)
+{
+    toAppend.append("<font color=\"" + format.color + "\">");
+    if (format.bold)
+        toAppend.append("<b>");
+    if (format.italic)
+        toAppend.append("<i>");
+    if (format.underlined)
+        toAppend.append("<u>");
+    if (format.strikethrough)
+        toAppend.append("<s>");
+}
+
+void appendEndFormat(TextFormat& format, QString& toAppend)
+{
+    toAppend.append("</font>");
+    if (format.bold)
+        toAppend.append("</b>");
+    if (format.italic)
+        toAppend.append("</i>");
+    if (format.underlined)
+        toAppend.append("</u>");
+    if (format.strikethrough)
+        toAppend.append("</s>");
+}
+
+bool processComponent(const QJsonValue& value, QString& result, TextFormat* parentFormat = nullptr);
+
+bool processComponentList(const QJsonArray& arr, QString& result, TextFormat* parentFormat = nullptr)
+{
+
+    for (const QJsonValue& val : arr) {
+        if (!processComponent(val, result, parentFormat))
+            return false;
+    }
+
+    return true;
+}
+
+bool processComponent(const QJsonValue& value, QString& result, TextFormat* parentFormat)
+{
+    if (value.isString()) {
+        result.append(value.toString());
+    } else if (value.isObject()) {
+        auto obj = value.toObject();
+
+        TextFormat format{};
+        if (parentFormat)
+            format = *parentFormat;
```

Comment:
Because you are translating things in HTML you do not need to pass the parentFormat to the next function.
You already surrounded the text with bold or italic you do not need to do that anymore.